src/Integration.UnitTests/ProjectToLanguageMapperTests.cs

```diff
@@ -0,0 +1,108 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    [TestClass]
+    public class ProjectToLanguageMapperTests
+    {
+        [TestMethod]
+        public void Mapper_ForProject_KnownLanguage_ArgChecks()
+        {
+            Exceptions.Expect<ArgumentNullException>(() => ProjectToLanguageMapper.GetLanguageForProject(null));
+        }
+
+        [TestMethod]
+        public void Mapper_ForProject_UnknownLanguage_ReturnsUnknown()
+        {
+            // Arrange
+            var otherProject = new ProjectMock("other.proj");
+
+            // Act
+            var otherProjectLanguage = ProjectToLanguageMapper.GetLanguageForProject(otherProject);
+
+            // Assert
+            otherProjectLanguage.Should().Be(Language.Unknown, "Unexpected Language for unknown project");
+        }
+
+        [TestMethod]
+        public void Mapper_ForProject_KnownLanguage_ReturnsCorrectLanguage_CS()
+        {
+            // Arrange
+            var csProject = new ProjectMock("cs1.csproj");
+            csProject.SetCSProjectKind();
+
+            // Act
+            var csProjectLanguage = ProjectToLanguageMapper.GetLanguageForProject(csProject);
+
+            // Assert
+            csProjectLanguage.Should().Be(Language.CSharp, "Unexpected Language for C# project");
+        }
+
+        [TestMethod]
+        public void Mapper_ForProject_KnownLanguage_ReturnsCorrectLanguage_VB()
+        {
+            // Test case 3: VB - non-Core
+            // Arrange
+            var vbNetProject = new ProjectMock("vb1.vbproj");
+            vbNetProject.SetVBProjectKind();
+
+            // Act
+            var vbNetProjectLanguage = ProjectToLanguageMapper.GetLanguageForProject(vbNetProject);
+
+            // Assert
+            vbNetProjectLanguage.Should().Be(Language.VBNET, "Unexpected Language for VB project");
+        }
+
+        [Ignore] // not yet implemented
+        [TestMethod]
+        public void Mapper_ForProject_KnownLanguage_ReturnsCorrectLanguage_CSCore()
+        {
+            // Arrange
+            var csProject = new ProjectMock("cs1.csproj");
+            csProject.ProjectKind = ProjectSystemHelper.CSharpCoreProjectKind;
+
+            // Act
+            var csProjectLanguage = ProjectToLanguageMapper.GetLanguageForProject(csProject);
+
+            // Assert
+            csProjectLanguage.Should().Be(Language.CSharp, "Unexpected Language for C# Core project");
+        }
+
+        [Ignore] // not yet implemented
```

Comment:
Also here, justification would be useful when we come back to it after n months.

Reply:
@michalb-sonar "not yet implemented" is the justification, and there's a TODO in the product code. What comment would you prefer to see, and where?

Reply:
Ah, ok. I assumed your [previos comment](https://github.com/SonarSource/sonarlint-visualstudio/pull/431/files#r150467145) means there is a 3rd party limitation - hence I said it's useful to say that. Guess that is not the case.